### PR TITLE
Fix PEP 695 output for classes in the LaTeX builder

### PR DESCRIPTION
**Pull request by picnixz**
Related: #12543.

@jbms Can you try if the PR solves the first issue? (not the one where you inject additional type parameters).

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -89,6 +89,8 @@ Features added
 Bugs fixed
 ----------
 
+* #12543: Fix :pep:`695` formatting for LaTeX output.
+  Patch by Bénédikt Tran.
```
Comment (AA-Turner) [inline CHANGES.rst]: This should go at the end of the list

File: sphinx/domains/python/_annotations.py [hunk 2]
```diff
@@ -265,7 +265,7 @@ def _build_identifier(self, tokens: list[Token]) -> str:
         from itertools import chain, islice
 
         def triplewise(iterable: Iterable[Token]) -> Iterator[tuple[Token, ...]]:
-            # sliding_window('ABCDEFG', 4) --> ABCD BCDE CDEF DEFG
+            # triplewise('ABCDEFG') --> ABC BCD CDE DEF EFG
```
Comment (AA-Turner) [inline sphinx/domains/python/_annotations.py]: Unrelated? (``sliding_window`` is the name for the generic, I named this one ``triplewise`` as it is a concrete version)

File: sphinx/writers/latex.py [hunk 3]
```diff
@@ -724,19 +724,21 @@ def has_multi_line(e: Element) -> bool:
             return e.get('multi_line_parameter_list')
 
         self.has_tp_list = False
+        self.orphan_tplist = False
```
Comment (AA-Turner) [inline sphinx/writers/latex.py]:
```suggestion
        self.orphan_tp_list = False
```

File: sphinx/writers/latex.py [hunk 4]
```diff
@@ -724,19 +724,21 @@ def has_multi_line(e: Element) -> bool:
...
                 multi_tp_list = has_multi_line(child)
-                multi_arglist = has_multi_line(arglist)
+                arglist = next_sibling(child)
+                if isinstance(arglist, addnodes.desc_parameterlist):
+                    # tp_list + arglist: \macro{name}{tp_list}{arglist}{retann}
+                    multi_arglist = has_multi_line(arglist)
+                else:
+                    # orphan tp_list:    \macro{name}{tp_list}{}{retann}
+                    # see: https://github.com/sphinx-doc/sphinx/issues/12543
+                    self.orphan_tplist = True
```
Comment (AA-Turner) [inline sphinx/writers/latex.py]:
```suggestion
                    self.orphan_tp_list = True
```

File: sphinx/writers/latex.py [hunk 5]
```diff
@@ -857,7 +859,13 @@ def _visit_sig_parameter_list(self, node: Element, parameter_group: type[Element
         self.multi_line_parameter_list = node.get('multi_line_parameter_list', False)
 
     def visit_desc_parameterlist(self, node: Element) -> None:
-        if not self.has_tp_list:
+        if self.has_tp_list:
+            if self.orphan_tplist:
```
Comment (AA-Turner) [inline sphinx/writers/latex.py]:
```suggestion
            if self.orphan_tp_list:
```

**Review by AA-Turner — COMMENTED** [review]
Minor points:

File: sphinx/domains/python/_annotations.py (hunk 2 above)
Comment (picnixz) [reply sphinx/domains/python/_annotations.py]: Ah! I thought it was me who forgot to update the comment! (but my triplewise was using pairwise instead). Will revert.

File: CHANGES.rst (hunk 1 above)
Comment (picnixz) [reply CHANGES.rst]: Oh I wasn't sure which was the last and which was the first :')

**Pull request merged**